### PR TITLE
[483] Added puma ssl/https and related items

### PR DESCRIPTION
**Pull request by JonF1**
### Context
This is cherry-picked from Sau:
https://github.com/DFE-Digital/register-trainee-teachers/pull/126

puma ssl/https and related items

### Changes proposed in this pull request

Added puma ssl/https and related items
default port from 3000 to 5000
zeroed ip (0.0.0.0)
default from HTTP to HTTPS
enforced ssl usage as default

### Guidance to review

Ported mainly from https://github.com/DFE-Digital/publish-teacher-training/blob/master/config/puma.rb

These are the minimum security and infrastructures requirements for future work.

Port 5000 was selected to remove common default rails port collusion.

Navigate to https://localhost:5000

If you see a certificate error message on Chrome on Mac - click and type 'thisisunsafe' (i'm not joking)

### CODE REVIEW

**Comment by defong** [comment]
[[M] Add a sign in page](https://trello.com/c/10W1RWyg/483-m-add-a-sign-in-page)

File: config/settings.yml [hunk 1]
```diff
@@ -9,3 +11,230 @@ dttp:
 # Used to add feature flags in the app to control access to certain features.
 features:
   home_text: false
+  use_ssl: true
+
+# Others
+nationalities:
```
Comment (darokel) [inline config/settings.yml:17]: I don't think this nationalities chunk should be in here. It may have leaked in

**Review by darokel — CHANGES_REQUESTED** [review]
Could we remove the nationalities list from the yaml, I don't think it should be in this PR.

**Review by darokel — APPROVED** [review]

File: config/settings.yml [hunk 2]
```diff
@@ -9,3 +11,5 @@ dttp:
 # Used to add feature flags in the app to control access to certain features.
 features:
   home_text: false
+  use_ssl: true
+
```
Comment (aliuk2012) [inline config/settings.yml:15]: end of file

Comment (defong) [reply config/settings.yml]: :+1:

**Pull request merged**